### ISSUE TITLE
False positive on unused enum members when using multiple entrypoints + other conditions

### ISSUE THREAD

**Issue by shawnmcknight** [webpro-nl/knip#703]
I've created a reproduction of this issue at https://github.com/shawnmcknight/knip-enum-member.

This one is really an edge case and I've tried to make the reproduction as minimal as possible. The minimal reproduction required the following:
- Create multiple modules with exported types with one of the types being an enumeration
- Have these modules import types from each other
- Have an entry point which re-exports the types
- Have another entry point which imports the enumeration from the re-exporting entry point
- Use the enumeration in the other entry point

With all of these conditions in place, there are unused exported enum members being reported, even though they are being used.

Reproduction code:
**fruit.ts**
```ts
import type { Tree } from "./tree";

export enum Fruit {
  apple = "apple",
  orange = "orange",
}

export interface Farmer {
  plants: Tree[];
}
```
**tree.ts**
```ts
import type { Fruit } from "./fruit";

export interface Tree {
  fruit: Fruit;
}
```
**index.ts**
```ts
export { Fruit, Farmer } from "./fruit";
export { Tree } from "./tree";
```
**main.ts**
```ts
import { Fruit } from "."; // shows unused exported enum members

const fruitBasket = {
  apple: Fruit.apple,
  orange: Fruit.orange,
};
```
**.knip.jsonc**
```jsonc
{
  "$schema": "https://unpkg.com/knip@5/schema-jsonc.json",
  "entry": ["index.ts", "main.ts"]
}
```

With this scenario, knip will output:
```
Unused exported enum members (2)
apple   Fruit  src/types/fruit.ts:4:3
orange  Fruit  src/types/fruit.ts:5:3
```
even though the enum members are being used in `main.ts`. An interesting side note here is that if `main.ts` imports from `./fruit` instead of from `.` then it will not report that these enum members have not been used. Although there were a lot of conditions to triggering this issue, I could not make a reproduction unless all of those aforementioned conditions were in place. If any of the conditions were not used then knip would correctly report no errors.

**Comment by webpro** [comment]
Thanks for the detailed report + repro! The fix was rather simple: don't skip enum members in entry files, even with `--include-entry-exports`

**Comment by shawnmcknight** [comment]
Wow, that was a really fast response! 🎉 
Thank you!

**Comment by webpro** [comment]
:rocket: _This issue has been resolved in v5.23.1. See [Release 5.23.1](https://github.com/webpro-nl/knip/releases/tag/5.23.1) for release notes._

_Using Knip in a commercial project? Please consider [becoming a sponsor](https://knip.dev/sponsors)._
